networkdb/networkdb.go

```diff
@@ -436,8 +436,7 @@ func (nDB *NetworkDB) deleteNetworkEntriesForNode(deletedNode string) {
 	nDB.Unlock()
 }
 
-func (nDB *NetworkDB) deleteNodeNetworkEntries(nid, node string) {
-	nDB.Lock()
+func (nDB *NetworkDB) deleteNodeNetworkEntries(nid, node string, flush bool) {
```

Comment:
The logic is very complicated depending on the case that is being handled by this function.
Can you please document this function clearly on which cases are addressed by each block ?
Also, the name `flush` is a bit misleading. Consider renaming it ?